### PR TITLE
[API] Add repoCreateTag

### LINKED ISSUE #14669: [API] Add repoCreateTag function
`POST /repos/{owner}/{repo}/tags` to create a tag

### PR DESCRIPTION
**Pull request by 6543**
`POST /repos/{owner}/{repo}/tags` to create a tag

close #14669

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: routers/api/v1/repo/tag.go [hunk 1]
```diff
@@ -160,3 +162,62 @@ func DeleteTag(ctx *context.APIContext) {
...
+		form.Target = ctx.Repo.Repository.DefaultBranch
+	}
+
+	commit, err := ctx.Repo.GitRepo.GetCommit(form.Target)
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "target not found", fmt.Errorf("target not found: %v", err))
+		return
+	}
+
+	if err := releaseservice.CreateNewTag(ctx.User, ctx.Repo.Repository, commit.ID.String(), form.TagName, form.Message); err != nil {
```
Comment (zeripath) [inline routers/api/v1/repo/tag.go:208]: Just double check that this matches the commit api.

Often we allow people to set a title and a message body separately which would be:

form.Title + "\n\n" + form.Message

Comment (6543) [reply routers/api/v1/repo/tag.go:208]: @zeripath this func is already in use by webUI tag creation ... so it should work fine

Comment (zeripath) [reply routers/api/v1/repo/tag.go:208]: Yes and in the web we artificially split the title and the message:

https://github.com/go-gitea/gitea/blob/1bfb0a24d843e10d6d95c4319a84980485e584ed/routers/web/repo/release.go#L311

Comment (zeripath) [reply routers/api/v1/repo/tag.go:208]: However file API.FileOptions doesn't appear to separate things:

https://github.com/go-gitea/gitea/blob/ab77a24f18ae45248bd6c71778fe3bfacacf864f/modules/structs/repo_file.go#L9-L11

So it's probably fine

Comment (6543) [reply routers/api/v1/repo/tag.go:208]: now I know what you ment ... yes this better as is for api ... make it consistent :)

File: templates/swagger/v1_json.tmpl [hunk 2]
```diff
@@ -17165,7 +17235,7 @@
     "parameterBodies": {
       "description": "parameterBodies",
       "schema": {
-        "$ref": "#/definitions/PullReviewRequestOptions"
+        "$ref": "#/definitions/CreateTagOption"
```
Comment (lunny) [inline templates/swagger/v1_json.tmpl:17338]: Is this wrong before?

Comment (6543) [reply templates/swagger/v1_json.tmpl:17338]: @lunny no this is a hack we use to generate swagger doc ...

Comment (6543) [reply templates/swagger/v1_json.tmpl:17338]: so it's alright

Comment (6543) [reply templates/swagger/v1_json.tmpl:17338]: @lunny do you need more info on that?

**Review by khmarbaise — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**